### PR TITLE
Fixes zonemaster-engine French translation

### PR DESCRIPTION
**Pull request by vlevigneron**
- Fixes zonemaster/zonemaster-engine#599
- Fixes zonemaster/zonemaster-engine#551
- Fixes zonemaster/zonemaster-engine#607
Applies only for fr.po.
Some DNSSEC messages missing, will be added soon

### CODE REVIEW

**Comment by mattias-p** [comment]
To fully fix #599 you must also remove fr.po from https://github.com/zonemaster/zonemaster-engine/blob/develop/t/po-files.t#L30.

File: share/fr.po [hunk 1]
```diff
@@ -235,24 +238,24 @@ msgstr ""
...
-#, fuzzy, perl-brace-format
+#, perl-brace-format
 msgid ""
 "Child does not list enough ({count}) nameservers that resolve to IPv6 "
 "addresses ({addrs}). Lower limit set to {minimum}."
 msgstr ""
 "Les serveurs de noms de la zone ne retournent pas assez de serveurs "
-"({count}) faisant autorité ({ns}). La limite inférieure étant fixée à "
-"{minimum}."
+"({count}) faisant autorité ({ns}) ayant une adresse IPv6 ({addrs}). La "
```
Comment (mattias-p) [inline share/fr.po]: There's no {ns} in the msgid.

File: share/fr.po [hunk 2]
```diff
@@ -235,24 +238,24 @@ msgstr ""
...
-#, fuzzy, perl-brace-format
+#, perl-brace-format
 msgid ""
 "Child does not list enough ({count}) nameservers that resolve to IPv4 "
 "addresses ({addrs}). Lower limit set to {minimum}."
 msgstr ""
 "Les serveurs de noms de la zone ne retournent pas assez de serveurs "
-"({count}) faisant autorité ({ns}). La limite inférieure étant fixée à "
-"{minimum}."
+"({count}) faisant autorité ({ns}) ayant une adresse IPv4 ({addrs}). La "
```
Comment (mattias-p) [inline share/fr.po]: There's no {ns} in the msgid.

File: share/fr.po [hunk 3]
```diff
@@ -281,38 +284,42 @@ msgstr ""
...
-#, fuzzy, perl-brace-format
+#, perl-brace-format
 msgid ""
 "Child lists enough ({count}) nameservers that resolve to IPv4 addresses "
 "({addrs}). Lower limit set to {minimum}."
 msgstr ""
 "Les serveurs de noms de la zone retournent suffisamment de serveurs "
-"({count}) faisant autorité ({ns}). La limite inférieure étant fixée à "
-"{minimum}."
+"({count}) faisant autorité ({ns}) ayant une adresse IPv4 ({addrs}). La "
```
Comment (mattias-p) [inline share/fr.po]: There's no {ns} in msgid.

File: share/fr.po [hunk 4]
```diff
@@ -281,38 +284,42 @@ msgstr ""
...
-#, fuzzy, perl-brace-format
+#, perl-brace-format
 msgid ""
 "Child lists enough ({count}) nameservers that resolve to IPv6 addresses "
 "({addrs}). Lower limit set to {minimum}."
 msgstr ""
 "Les serveurs de noms de la zone retournent suffisamment de serveurs "
-"({count}) faisant autorité ({ns}). La limite inférieure étant fixée à "
-"{minimum}."
+"({count}) faisant autorité ({ns}) ayant une adresse IPv6 ({addrs}). La "
```
Comment (mattias-p) [inline share/fr.po]: There's no {ns} in msgid.

**Comment by vlevigneron** [comment]
@mattias-p OK, I'm on it.

File: share/fr.po [hunk 5]
```diff
@@ -342,39 +349,25 @@ msgstr ""
...
+#, perl-brace-format
 msgid ""
 "Delegation does not list enough ({count}) nameservers that resolve to IPv4 "
 "addresses ({addrs}). Lower limit set to {minimum}."
 msgstr ""
-"Les serveurs de noms de la zone ne retournent pas assez de serveurs "
-"({count}) faisant autorité ({ns}). La limite inférieure étant fixée à "
-"{minimum}."
+"Les serveurs de noms de la zone parente ne retournent pas assez de serveurs "
+"({count}) faisant autorité ({ns}) ayant une adresse IPv4 ({addrs}). La "
```
Comment (mattias-p) [inline share/fr.po]: There's no {ns} in the msgid.

File: share/fr.po [hunk 6]
```diff
@@ -342,39 +349,25 @@ msgstr ""
...
+#, perl-brace-format
 msgid ""
 "Delegation does not list enough ({count}) nameservers that resolve to IPv6 "
 "addresses ({addrs}). Lower limit set to {minimum}."
 msgstr ""
-"Les serveurs de noms de la zone ne retournent pas assez de serveurs "
-"({count}) faisant autorité ({ns}). La limite inférieure étant fixée à "
-"{minimum}."
+"Les serveurs de noms de la zone parente ne retournent pas assez de serveurs "
+"({count}) faisant autorité ({ns}) ayant une adresse IPv6 ({addrs}). La "
```
Comment (mattias-p) [inline share/fr.po]: There's no {ns} in the msgid.

File: share/fr.po [hunk 7]
```diff
@@ -388,38 +381,42 @@ msgid "Delegation from parent to child is properly signed."
...
+#, perl-brace-format
 msgid ""
 "Delegation lists enough ({count}) nameservers that resolve to IPv4 addresses "
 "({addrs}). Lower limit set to {minimum}."
 msgstr ""
-"Les serveurs de noms de la zone retournent suffisamment de serveurs "
-"({count}) faisant autorité ({ns}). La limite inférieure étant fixée à "
-"{minimum}."
+"Les serveurs de noms de la zone parente retournent suffisamment de serveurs "
+"({count}) faisant autorité ({ns}) ayant une adresse IPv4 ({addrs}). La "
```
Comment (mattias-p) [inline share/fr.po]: There's no {ns} in the msgid.

File: share/fr.po [hunk 8]
```diff
@@ -388,38 +381,42 @@ msgid "Delegation from parent to child is properly signed."
...
+#, perl-brace-format
 msgid ""
 "Delegation lists enough ({count}) nameservers that resolve to IPv6 addresses "
 "({addrs}). Lower limit set to {minimum}."
 msgstr ""
-"Les serveurs de noms de la zone retournent suffisamment de serveurs "
-"({count}) faisant autorité ({ns}). La limite inférieure étant fixée à "
-"{minimum}."
+"Les serveurs de noms de la zone parente retournent suffisamment de serveurs "
+"({count}) faisant autorité ({ns}) ayant une adresse IPv6 ({addrs}).  La "
```
Comment (mattias-p) [inline share/fr.po]: There's no {ns} in the msgid.
There's a double space.

**Comment by vlevigneron** [comment]
@mattias-p I'm working on missing french translations. For {ns}, the issue is in the code and en.po. {ns} (that should be changed into something like {nss}) is set in the code but not used in english version of messages. I will make a dedicated issue for that case.

File: share/fr.po (hunk 1 above)
Comment (vlevigneron) [reply share/fr.po]: Will be fixed in zonemaster/zonemaster-engine#648

File: share/fr.po (hunk 2 above)
Comment (vlevigneron) [reply share/fr.po]: Will be fixed in zonemaster/zonemaster-engine#648

File: share/fr.po (hunk 3 above)
Comment (vlevigneron) [reply share/fr.po]: Will be fixed in zonemaster/zonemaster-engine#648

File: share/fr.po (hunk 5 above)
Comment (vlevigneron) [reply share/fr.po]: Will be fixed in zonemaster/zonemaster-engine#648

File: share/fr.po (hunk 6 above)
Comment (vlevigneron) [reply share/fr.po]: Will be fixed in zonemaster/zonemaster-engine#648

File: share/fr.po (hunk 4 above)
Comment (vlevigneron) [reply share/fr.po]: Will be fixed in zonemaster/zonemaster-engine#648

File: share/fr.po (hunk 7 above)
Comment (vlevigneron) [reply share/fr.po]: Will be fixed in zonemaster/zonemaster-engine#648

File: share/fr.po (hunk 8 above)
Comment (vlevigneron) [reply share/fr.po]: Will be fixed in zonemaster/zonemaster-engine#648

**Comment by vlevigneron** [comment]
@mattias-p DNSSEC:DS_ALGORITHM_OK and DNSSEC:DS_ALGORITHM_MISSING are left untranslated on purpose. It will be adressed in zonemaster/zonemaster-engine#646

File: share/fr.po [hunk 9]
```diff
@@ -757,20 +764,27 @@ msgid ""
 "query that is expected to give response with RCODE \"NXDOMAIN\". Test for "
 "NSEC and NSEC3 is aborted for this nameserver."
 msgstr ""
+"Le serveur de noms {ns}/{address} pour la zone {zone} répond avec un code "
```
Comment (mattias-p) [inline share/fr.po]: "{ns}/{address}" -> "{ns/address}"

File: share/fr.po [hunk 10]
```diff
@@ -757,20 +764,27 @@ msgid ""
...
+"retour \"NOERROR\" alors que le code retour \"NXDOMAIN\" était attendu. La "
+"suite des tests \"NSEC\" et \"NSEC3\" est avortée pour ce serveur de nom."
 
 #. DNSSEC:MIXED_NSEC_NSEC3
 #, perl-brace-format
 msgid ""
 "Nameserver {ns/address} for zone {zone} responds with both NSEC and NSEC3 "
 "records when only one record type is expected."
 msgstr ""
+"Le serveur de noms {ns}/{address} pour la zone {zone} répond avec un mélange "
```
Comment (mattias-p) [inline share/fr.po]: Ditto.

File: share/fr.po [hunk 11]
```diff
@@ -757,20 +764,27 @@ msgid ""
...
+"d'enregistrements de types \"NSEC\" et \"NSEC3\" alors qu'un seul type est attendu."
 
 #. DNSSEC:NO_NSEC_NSEC3
 #, perl-brace-format
 msgid ""
 "Nameserver {ns/address} for zone {zone} responds with neither NSEC nor NSEC3 "
 "record when when such records are expected."
 msgstr ""
+"Le serveur de noms {ns}/{address} pour la zone {zone} ne répond avec aucun des "
```
Comment (mattias-p) [inline share/fr.po]: Ditto.

File: share/fr.po (hunk 9 above)
Comment (vlevigneron) [reply share/fr.po]: Good catch but the issue is in DNSSEC.pm where I would prefer do the fix.
I propose not to fix it in this PR but in zonemaster/zonemaster-engine#646

File: share/fr.po (hunk 10 above)
Comment (vlevigneron) [reply share/fr.po]: Same comment

File: share/fr.po (hunk 11 above)
Comment (vlevigneron) [reply share/fr.po]: Same comment

File: share/fr.po (hunk 9 above)
Comment (mattias-p) [reply share/fr.po]: Fair enough.

**Pull request merged**